### DISCUSSION TITLE
Automatic discovery of multiple sites on the same server: ability to serialize update and backup operations automatically

### DISCUSSION THREAD

**Discussion by morphinestyle** [akeeba/panopticon#527] [Ideas]
What happens when I start updates or backups on multiple sites that are on the same server? (as in my case)
Is there a risk of KOing the server?

A feature that avoids loading the server during operations, perhaps by serializing them, would be useful.

**Comment by nikosdion** [comment]
Generally, we say that you shouldn't do ‘too much’ on your server at once. The amount that's ‘too much’ depends on the server.

From lightest to heaviest operation in terms of server resource usage it goes extension updates, Joomla update, backup, PHP File Change Scanner.

The mechanism to avoid overlap is simple. First, you can configure the time of day core updates, backups, and scans take place. Make sure they are not all set for the exact same time. Space them out. How? It's your server, it's your sites, you get to decide. Second, Panopticon has a finite capacity of simultaneous execution, limited by the number of CRON jobs. One CRON job can do up to three things at once (assuming every minute execution, a time limit of 180 seconds, and use of the --loop parameter). Therefore, the limited capacity will prevent you from overdoing it.

At some point you MAY have too many sites to not have any significant overlap. You know what that means? You're well beyond the point you should have been using a different server to add new sites.

So, no need to create anything overcomplicated. Yes, I did have the same thought as you did, then I started breaking it down to what it means, and I figured out that because of the above a feature like this would be practically unnecessary.
